src/azul/plugins/metadata/hca/transform.py

```diff
@@ -1480,6 +1498,7 @@ def _transform(self, cell_suspensions: Iterable[api.CellSuspension]) -> Iterable
                             sequencing_processes=list(
                                 map(self._sequencing_process, visitor.sequencing_processes.values())
                             ),
+                            dates=[self._date(self._entity(cell_suspension))],
```

Comment:
The fact that we call `self._entity()` here is a smell. `_entity` defines the properties that all inner entities share. Calling it here makes the somewhat brittle assumption that `_entity` adds exactly the properties you need and no more.

More importantly, the fact that `_entity` still adds the dates seems to waste a lot of space. Correct me if I am wrong, but I don't think we need to track the date of every inner entity if we track the dates of the outer entity in the `dates` inner entity.